src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Html/HtmlProjectedDocument.ts

```diff
@@ -25,19 +26,42 @@ export class HtmlProjectedDocument implements IProjectedDocument {
         return this.projectedDocumentVersion;
     }
 
+    public update(edits: ServerTextChange[], hostDocumentVersion: number) {
+        this.hostDocumentVersion = hostDocumentVersion;
+
+        if (edits.length === 0) {
+            return;
+        }
+
+        let content = this.content;
+        for (const edit of edits) {
+            // TODO: Use a better data structure to represent the content, string concatenation is slow.
+            content = this.getEditedContent(edit.newText, edit.span.start, edit.span.end, content);
+        }
+
+        this.setContent(content);
+    }
+
     public getContent() {
         return this.content;
     }
 
-    public setContent(content: string, hostDocumentVersion: number) {
+    public reset() {
         this.projectedDocumentVersion++;
-        this.hostDocumentVersion = hostDocumentVersion;
-        this.content = content;
+        this.hostDocumentVersion = null;
+        this.setContent('');
```

Comment:
This increments the projected document version is that what you expect?

Reply:
Yes, resetting should increment version. We do the same thing in `CSharpProjectedDocument`.